File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/utils/DeviceUtils.java
package com.neatorobotics.sdk.android.utils;

import android.os.Build;

/**
 * Neato-SDK
 * Created by Marco on 06/05/16.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */
public class DeviceUtils {

    public static String getModel(){
        return Build.MODEL;
    }
    public static String getManufacturer(){
        return Build.MANUFACTURER;
    }
    public static int getAndroidVersionAPI() {
        return Build.VERSION.SDK_INT;
    }

    public static String getXAgentString() {
        StringBuffer builder = new StringBuffer();
        builder.append("android-");
        builder.append(DeviceUtils.getAndroidVersionAPI());
        builder.append("|");
        builder.append(DeviceUtils.getModel());
        builder.append("|");
        builder.append(DeviceUtils.getManufacturer());
        return builder.toString();
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/utils/DeviceUtils.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.utils

import android.os.Build

object DeviceUtils {

    val model: String
        get() = Build.MODEL
    val manufacturer: String
        get() = Build.MANUFACTURER
    val androidVersionAPI: Int
        get() = Build.VERSION.SDK_INT

    val xAgentString: String
        get() {
            val builder = StringBuffer()
            builder.append("android-")
            builder.append(DeviceUtils.androidVersionAPI)
            builder.append("|")
            builder.append(DeviceUtils.model)
            builder.append("|")
            builder.append(DeviceUtils.manufacturer)
            return builder.toString()
        }
}
